pkg/orchestrator/orchestrator.go

```diff
@@ -100,12 +100,17 @@ func New(ctx context.Context, config *Config) (*Orchestrator, error) {
 		return nil, fmt.Errorf("failed to create a backend client: %w", err)
 	}
 
-	p, err := NewProvider(ctx, config.ProviderKind)
+	// TODO(paralta) Provider initialization needs to be removed from here once Providers are split from Orchestrator (Issue #643).
+	apiProvider, err := backendClient.PostProvider(ctx, config.Provider)
 	if err != nil {
-		return nil, fmt.Errorf("failed to initialize provider. Provider=%s: %w", config.ProviderKind, err)
+		return nil, fmt.Errorf("failed to post provider: %w", err)
+	}
+	p, err := NewProvider(ctx, apiProvider)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize provider. Provider=%s: %w", *apiProvider.DisplayName, err)
 	}
```

Comment:
@chrisgacsal @paralta I think we need to chat about the initial registration part to avoid creating new providers on orchestrator restart. i.e. we could use passed displayName to check to see if the provider already exists in the API and if it does save that ID for use in the rest of the flow. Alternatively check to see if /etc/vmclarity/providerID exists, and if it does not we POST a new provider and then save the ID into that file. If the file does exist then we PATCH the display name etc.

I think the later might make more sense in the long term but does require persistent storage of some kind (this could be different per provider i.e. configmap for kubernetes, file for VM on AWS, volume for docker. In the future we'll probably need to store some kind of auth certificate anyway.